app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -115,6 +117,20 @@ private void extractIntentFields(Intent intent)
                 + ", updateLoyaltyCard=" + updateLoyaltyCard);
     }
 
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle savedInstanceState) {
+        super.onSaveInstanceState(savedInstanceState);
+        tabs = findViewById(R.id.tabs);
+        savedInstanceState.putInt(STATE_TAB_INDEX, tabs.getSelectedTabPosition());
+    }
+
+    @Override
+    public void onRestoreInstanceState(@Nullable Bundle savedInstanceState) {
```

Comment:
This should be NonNull because Android documentation states it will never be called with null and to make spotBugs happy.